src/dataflow/src/source/metrics.rs

```diff
@@ -75,7 +75,7 @@ impl KinesisMetrics {
             millis_behind_latest: registry.register(metric!(
                 name: "mz_kinesis_shard_millis_behind_latest",
                 help: "How far the shard is behind the tip of the stream",
-                var_labels: ["stream_name", "shard_id"],
+                var_labels: ["worker_id", "source_id", "stream_name", "shard_id"],
```

Comment:
I forgot to mention it yesterday but beware the "worker" label on histograms: It can get difficult to aggregate them into a common, global histogram.

Reply:
(This concrete one here is a counter, but if we have any histogram metrics, we should check if that results in any trouble)

Reply:
I reverted the changes for the particular metric. I think it requires some design work on how to handle metrics that might be written by multiple workers while avoiding the concurrent deletion problem; it's not something I want to touch as part of this PR.